test/conformance/usm/urUSMFree.cpp

```diff
@@ -95,3 +95,84 @@ TEST_P(urUSMFreeTest, InvalidNullPtrMem) {
     ASSERT_EQ_RESULT(UR_RESULT_ERROR_INVALID_NULL_POINTER,
                      urUSMFree(context, nullptr));
 }
+
+// This goal of this test is to ensure urUSMFree blocks and waits for operations
+// accessing the given allocation to finish before actually freeing the memory.
+struct urUSMFreeDuringExecutionTest : uur::urKernelExecutionTest {
+    void SetUp() {
+        program_name = "fill_usm";
+        UUR_RETURN_ON_FATAL_FAILURE(urKernelExecutionTest::SetUp());
+    }
+
+    void *allocation = nullptr;
+    size_t array_size = 256;
+    size_t allocation_size = array_size * sizeof(uint32_t);
+    uint32_t data = 42;
+    size_t wg_offset = 0;
+};
+UUR_INSTANTIATE_KERNEL_TEST_SUITE_P(urUSMFreeDuringExecutionTest);
```

Comment:
When I run `ctest -R usm` locally it works fine but when I run the binary directly I get:

```
igchor@DUT1044PVC:~/unified-runtime/build$ UR_ADAPTERS_FORCE_LOAD=./lib/libur_adapter_level_zero.so ./bin/test-usm
Selected platform: [LEVEL_ZERO] Intel(R) oneAPI Unified Runtime over Level-Zero
Segmentation fault (core dumped)
```

I belive this test should be in a separate .cpp file that is added using add_conformance_test_with_kernels_environment. urUSMFree.cpp is added with add_conformance_test_with_devices_environment so there's no kernels env available here and hence the segfault.

Reply:
thanks for testing and noticing that, I've enabled the kernel environment for the usm suite in the build system now so it should be fixed